### PR TITLE
Refactor JS

### PR DESCRIPTION
**Pull request by jermspeaks**
- Reset `cities.js`
- Leave `Gemlock.lock` alone
- Sublime indentation to spaces

Previously:
- Initialized variables on top for performance
- Used function form for every file
- Refactored massive functions into smaller, manageable functions (for readability and clarity)
- Added comments to all functions
- Note: not entirely tested. Couldn't figure out how to do post-email button interactions in DEV environment. Tested everything else out in Chrome. Unsure about Firefox, Safari and IE.

### CODE REVIEW

**Comment by mehulkar** [comment]
thanks man. But it looks like your PR reverted a lot of changes that were made in master and are not related to this PR. (See the "Files changed" tab to see what I'm talking about. e.g. all the `forbes_city_path` removals from #519 has been added back in here).

File: app/assets/javascripts/modals.js [hunk 1]
```diff
@@ -1,44 +1,88 @@
...
+
+  /**
+   * Activate Modal Callbacks
+   */
+  modalActivation = function() {
+    // $('.sign-up-emphasis').on('click', function(evt) {
+    //   if ($(evt.currentTarget).attr('href') === '/signup') {
+    //     modal = loadModal('/signup')(evt);
+    //   }
+    // });
```
Comment (mehulkar) [inline app/assets/javascripts/modals.js]: let's not comment out code. let's delete it entirely if it's meant to be deleted.

Comment (jermspeaks) [inline app/assets/javascripts/modals.js]: Thanks, consider it done.

**Comment by jermspeaks** [comment]
Is there a way to diff the files that were changed from the latest pull and remove them on the PR? I didn't mean to change all of the other files as well.

**Comment by mehulkar** [comment]
@jermspeaks picked out your work and put it in #533. can you verify that that is what you want?

**Comment by jermspeaks** [comment]
Yeah, that's it. I was having a few issues pulling upstream earlier. Thanks.

**Pull request closed without merging**